How to build and run this on Mac OS X?

I followed the steps described by @rsanheim and still got this two errors
```
/Users/sa/repos/xmr-stak-cpu/socket.cpp:196:8: error: no matching function for call to 'SSL_CTX_new'
        ctx = SSL_CTX_new(method);
              ^~~~~~~~~~~
/usr/include/openssl/ssl.h:1362:10: note: candidate function not viable: 1st argument ('const SSL_METHOD *' (aka 'const ssl_method_st *')) would lose const qualifier
SSL_CTX *SSL_CTX_new(SSL_METHOD *meth) DEPRECATED_IN_MAC_OS_X_VERSION_10_7_AND_LATER;
         ^
/Users/sa/repos/xmr-stak-cpu/socket.cpp:202:82: error: use of undeclared identifier 'SSL_OP_NO_COMPRESSION'
                SSL_CTX_set_options(ctx, SSL_OP_NO_SSLv2 | SSL_OP_NO_SSLv3 | SSL_OP_NO_TLSv1 | SSL_OP_NO_COMPRESSION);
                                                                                               ^
/usr/include/openssl/ssl.h:582:39: note: expanded from macro 'SSL_CTX_set_options'
        SSL_CTX_ctrl((ctx),SSL_CTRL_OPTIONS,(op),NULL)
```

--- a/minethd.cpp
+++ b/minethd.cpp
@@ -51,7 +51,7 @@ void thd_setaffinity(std::thread::native_handle_type h, uint64_t cpu_id)
 {
 #if defined(__APPLE__)
 	thread_port_t mach_thread;
-	thread_affinity_policy_data_t policy = { cpu_id };
+	thread_affinity_policy_data_t policy = { static_cast<integer_t>(cpu_id) };
 	mach_thread = pthread_mach_thread_np(h);
 	thread_policy_set(mach_thread, THREAD_AFFINITY_POLICY, (thread_policy_t)&policy, 1);
 #elif defined(__FreeBSD__)

